content/asp.net/getting-started/mvc.md

```diff
@@ -1,20 +1,122 @@
+>
+>We will use Visual Studio 2015 for these exercises. Any version will work provided you are running the latest version.
 
-...
+## Creating Your Project 
+Open up Visual Studio 2015 and create a new project from the File -> New -> Project menu. Under the templates selection, choose Web, then pick ASP.NET Core Web Application (.NET Core) from the list of available templates. Name your project "MvcBasics".
 
+![Select your project template](images\visual-studio-project-template.png)
+
+Press ``Ctrl+F5`` or choose Debug -> Start Without Debugging from the menu. You should see your application open in your default browser. 
+
+![Running your app for the first time](images\first-run.png)
+
+## MVC Configuration
+The differentiator between a stock ASP.NET application and an MVC-based app are the services and facilities that MVC Core provides. In order to make these services available to the runtime and to our own services we need to first configure them as our application initializes. Here is the code for the ``ConfigureServices`` method in ``startup.cs`` which was included as part of the project template:
+
+```c#
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Add framework services.
+        services.AddMvc();
+    }
+```
+
+``AddMvc`` is an extension method of ``IServiceCollection`` that gives MVC access to the service container and configures it for use during request processing.  Next, we have the ``Configure`` method, which is the appropriate place to add our middleware:
+
+```c#
+    public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+    {
+        loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+        loggerFactory.AddDebug();
+
+        if (env.IsDevelopment())
+        {
+            app.UseDeveloperExceptionPage();
+            app.UseBrowserLink();
+        }
+        else
+        {
+            app.UseExceptionHandler("/Home/Error");
+        }
+
+        app.UseStaticFiles();
+
+        app.UseMvc(routes =>
+        {
+            routes.MapRoute(
+                name: "default",
+                template: "{controller=Home}/{action=Index}/{id?}");
+        });
+    }
+
+```
+
+In the first part of this block you'll find some code that sets up the logging middleware, exception handling and finally the static files [middleware](static-files.md). Wrapping up the method is the call to ``UseMvc``, an extention method to ``IApplicationBuilder`` that configures the MVC middleware. MVC should be the last thing you provide, as it incurs the cost of processing a request and would add latency to things like static files. Keep this in mind as you order your middleware, and make sure that ``UseStaticFiles`` appears before the MVC middleware.  
```

Comment:
If you link only "middleware", it looks like the link is going to be about middleware in general. A link to static-files.md should probably be for the whole "static files middleware" (i.e. `the [static files middleware](static-files.md)`).